src/backend/access/transam/xact.c

```diff
@@ -2317,14 +2317,14 @@ CommitTransaction(void)
 	willHaveObjectsFromSmgr =
 			PersistentEndXactRec_WillHaveObjectsFromSmgr(EndXactRecKind_Commit);
 
-	if (willHaveObjectsFromSmgr)
-	{
-		/*
-		 * We need to ensure the recording of the [distributed-]commit record and the
-		 * persistent post-commit work will be done either before or after a checkpoint.
-		 */
-		CHECKPOINT_START_LOCK;
-	}
+	/* In previous version, we ensured the recording of the [distributed-]commit record and the
+	 * persistent post-commit work will be done either before or after a checkpoint.
+	 *
+	 * However the persistent table status will be synchronized with AOSeg_XXXX
+	 * table and hdfs file in PersistentRecovery_Scan() at recovery PASS2.
+	 * We don't need to worry about inconsistent states between them. So no
+	 * CHECKPOINT_START_LOCK any more.
+	 */
```

Comment:
The status of current transaction will not be truncated at (4), the truncate related code is:
CreateCheckPoint() --> CheckPointGuts() --> CheckPointMultiXact() -> TruncateMultiXact() -> SimpleLruTruncate()
In this function, cutoffPage is based on the min value of OldestMemberMXactId and OldestVisibleMXactId for all backends' proc. 

And OldestMemberMXactId and OldestVisibleMXactId for current backend's proc are reset in
AtEOXact_MultiXact(). In AbortTransaction() and CommitTransaction()， this function is called after AtEOXact_smgr(). 

So in recovery process, we can know that the current transaction has already committed, and all persistent table and hdfs files should be redo.